### PR TITLE
NXP-27711: add Jenkinsfile

### PR DESCRIPTION
**Pull request by troger**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      steps {
+        setGitHubBuildStatus('jsfui/utests/dev', 'Unit tests - dev environment', 'PENDING')
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Run "dev" unit tests
+          ----------------------------------------"""
+          withEnv(["MAVEN_OPTS=$MAVEN_OPTS -Xms512m -Xmx3072m"]) {
+            echo "MAVEN_OPTS=$MAVEN_OPTS"
+            sh "mvn -B test"
```
Comment (kevinleturc) [inline Jenkinsfile]: Shouldn't we need to add `-f code/pom.xml` ?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+      steps {
+        setGitHubBuildStatus('jsfui/ftests/dev', 'Functional tests - dev environment', 'PENDING')
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Run "dev" unit tests
+          ----------------------------------------"""
+          withEnv(["MAVEN_OPTS=$MAVEN_OPTS -Xms512m -Xmx3072m"]) {
+            echo "MAVEN_OPTS=$MAVEN_OPTS"
+            sh "mvn -B -nsu -f ftests/pom.xml test"
```
Comment (kevinleturc) [inline Jenkinsfile]: Should we generalize the `-nsu` option for others? (like utests)

**Review by kevinleturc — DISMISSED** [review]

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        container('maven') {
+          sh """
+            # to be removed when running with jenkins user
+            echo '{ "allow_root": true }' > /home/jenkins/.bowerrc
+          """
+        }
+      }
+    }
+
+    stage('Compile and install') {
```
Comment (ataillefer) [inline Jenkinsfile]: "Compile" to be consistent with the platform pipeline

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      steps {
+        setGitHubBuildStatus('jsfui/compile', 'Compile and install', 'PENDING')
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Compile and install
+          ----------------------------------------"""
+          withEnv(["MAVEN_OPTS=$MAVEN_OPTS -Xms512m -Xmx3072m"]) {
+            echo "MAVEN_OPTS=$MAVEN_OPTS"
+            sh 'mvn -B -nsu -T0.8C -f code/pom.xml install -DskipTests=true'
```
Comment (ataillefer) [inline Jenkinsfile]: `package` should be enough and slightly faster than `install`

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+      }
+    }
+
+    stage('Compile and install') {
+      steps {
+        setGitHubBuildStatus('jsfui/compile', 'Compile and install', 'PENDING')
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Compile and install
```
Comment (ataillefer) [inline Jenkinsfile]: Compile

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+        success {
+          setGitHubBuildStatus('jsfui/utests/dev', 'Unit tests - dev environment', 'SUCCESS')
+        }
+        failure {
+          setGitHubBuildStatus('jsfui/utests/dev', 'Unit tests - dev environment', 'FAILURE')
+        }
+      }
+    }
+
+    stage('Package and install') {
```
Comment (ataillefer) [inline Jenkinsfile]: Package

File: Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+      steps {
+        setGitHubBuildStatus('jsfui/package', 'Package and install', 'PENDING')
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Package
+          ----------------------------------------"""
+          withEnv(["MAVEN_OPTS=$MAVEN_OPTS -Xms512m -Xmx3072m"]) {
+            echo "MAVEN_OPTS=$MAVEN_OPTS"
+            sh 'mvn -B -nsu -f packages/pom.xml install -DskipTests=true'
```
Comment (ataillefer) [inline Jenkinsfile]: Can add `-T0.8C` and use `package` instead of `install`.

**Review by ataillefer — DISMISSED** [review]

**Comment by troger** [comment]
Will be done as part of https://github.com/nuxeo/nuxeo-jsf-ui/pull/19.

**Pull request closed without merging**